core/interfaces/src/application.rs

```diff
@@ -48,7 +58,7 @@ pub trait ApplicationInterface<C: Collection>:
     }
 
     /// The type for the sync query executor.
-    type SyncExecutor: SyncQueryRunnerInterface;
+    type SyncExecutor: QueryRunnerExt;
```

Comment:
This should be the core interface
```suggestion
    type SyncExecutor: SyncQueryRunnerInterface;
```

We should depend on the main one in generics/associated, and just use the extension in the consumers of the query runner